frontend/packages/topology/src/components/quick-search/QuickSearchList.tsx

```diff
@@ -0,0 +1,117 @@
+import { history, removeQueryArgument } from '@console/internal/components/utils';
+import { CatalogItem } from '@console/plugin-sdk';
+import {
+  DataList,
+  DataListItem,
+  DataListItemRow,
+  DataListItemCells,
+  DataListCell,
+  Split,
+  SplitItem,
+  Label,
+  TextContent,
+  Text,
+  TextVariants,
+  Button,
+} from '@patternfly/react-core';
+import './QuickSearchList.scss';
+
+interface QuickSearchListProps {
+  listItems: CatalogItem[];
+  totalItems: number;
+  selectedItemId: string;
+  searchTerm: string;
+  namespace: string;
+  onSelectListItem: (itemId: string) => void;
+}
+
+const QuickSearchList: React.FC<QuickSearchListProps> = ({
+  listItems,
+  totalItems,
+  selectedItemId,
+  searchTerm,
+  namespace,
+  onSelectListItem,
+}) => {
+  const { t } = useTranslation();
+  const ref = React.useRef(null);
+
+  const openForm = (e: React.SyntheticEvent) => {
+    e.persist();
+    e.stopPropagation();
+    e.preventDefault();
+    const targetItemUID = ref?.current?.props?.id;
+    const item = listItems.find((listItem) => listItem.uid === targetItemUID);
+    removeQueryArgument('catalogSearch');
+    history.push(item?.cta?.href);
+  };
+
+  const goToCatalogPage = () => {
+    removeQueryArgument('catalogSearch');
+    history.push(`/catalog/ns/${namespace}?keyword=${searchTerm}`);
+  };
+
+  return (
+    <DataList
+      className="odc-quick-search-list"
+      aria-label={t('topology~Quick Search List')}
+      selectedDataListItemId={selectedItemId}
+      onSelectDataListItem={(itemId) => itemId !== 'viewAll' && onSelectListItem(itemId)}
+      isCompact
+    >
+      {listItems.map((item) => (
+        <DataListItem
+          id={item.uid}
+          key={item.uid}
+          className={cx('odc-quick-search-list__item', {
+            'odc-quick-search-list__item--highlight': item.uid === selectedItemId,
+          })}
+          onDoubleClick={openForm}
+          ref={ref}
+        >
+          <DataListItemRow className="odc-quick-search-list__item-row">
+            <DataListItemCells
+              className="odc-quick-search-list__item-content"
+              dataListCells={[
+                <DataListCell isIcon>
+                  <img className="odc-quick-search-list__item-icon" src={item.icon?.url} alt="" />
+                </DataListCell>,
+                <DataListCell style={{ paddingTop: '6px' }} width={2} wrapModifier="truncate">
+                  <span className="odc-quick-search-list__item-name">{item.name}</span>
+                  <Split style={{ alignItems: 'center' }} hasGutter>
+                    <SplitItem>
+                      <Label>{item.type}</Label>
+                    </SplitItem>
+                    <SplitItem>
+                      <TextContent>
+                        <Text component={TextVariants.small}>{item.provider}</Text>
+                      </TextContent>
+                    </SplitItem>
+                  </Split>
+                </DataListCell>,
```

Comment:
PF looks to need us to provide keys on each one... perhaps combine UID + a static string?

From the console:
![image](https://user-images.githubusercontent.com/8126518/100369694-3e9ccb00-2fd3-11eb-9c7a-f2383904420b.png)

From PF docs:
![image](https://user-images.githubusercontent.com/8126518/100369745-51170480-2fd3-11eb-94e2-eedab17cc7df.png)